openid-connect-server/src/main/java/org/mitre/openid/connect/web/EndSessionValidator.java

```diff
@@ -0,0 +1,78 @@
+package org.mitre.openid.connect.web;
+
+import static org.springframework.util.StringUtils.isEmpty;
+
+import java.text.ParseException;
+
+import org.mitre.oauth2.model.OAuth2AccessTokenEntity;
+import org.mitre.oauth2.service.OAuth2TokenEntityService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.common.exceptions.InvalidTokenException;
+import org.springframework.stereotype.Component;
+
+import com.nimbusds.jwt.SignedJWT;
+
+@Component
+public class EndSessionValidator {
+	
+	private static final Logger logger = LoggerFactory.getLogger(EndSessionValidator.class);
+	
+	@Autowired
+	private OAuth2TokenEntityService tokenService;
+
+	public boolean isValid(String idTokenHint, String postLogoutRedirectUri, AbstractAuthenticationToken auth) {
+		if (idTokenHint == null || !isValidSyntax(idTokenHint) || isEmpty(postLogoutRedirectUri)) {
+			return false;
+		}
+		OAuth2AccessTokenEntity accessToken = findAccessToken(idTokenHint);
+		if (accessToken == null || 
+				accessToken.getClient() == null) {
+			return false;
+		} else if (accessToken.getClient().getPostLogoutRedirectUris() == null ||
+				!accessToken.getClient().getPostLogoutRedirectUris().contains(postLogoutRedirectUri)) {
+			logger.info("Unregistered post_logout_redirect_uri: " + postLogoutRedirectUri);
+			return false;
+		} else if (accessToken.getAuthenticationHolder() == null ||
+				accessToken.getAuthenticationHolder().getUserAuth() == null ||
+				accessToken.getAuthenticationHolder().getUserAuth().getName() == null ||
+				auth == null ||
+				auth.getPrincipal() == null ||
+				!accessToken.getAuthenticationHolder().getUserAuth().getName().equals(auth.getPrincipal())) {
```

Comment:
Shouldn't this be .equals(auth.getName())? You want to make sure the two names match, and even though the principal, which is an Object, could be a String with the name, it may not. Notice how the getName method in auth will return the principal (that your code uses directly) if it is not a UserDetails or Principal, so in your specific case it would not have any negative side-effects.